azext_edge/tests/edge/orchestration/test_upgrade2_unit.py

```diff
@@ -0,0 +1,413 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+import json
+import re
+from typing import Dict, List, Optional, Tuple, TypeVar
+from unittest.mock import Mock
+
+import pytest
+import requests
+import responses
+from azure.cli.core.azclierror import ValidationError
+from azure.core.exceptions import HttpResponseError
+
+from azext_edge.edge.providers.orchestration.common import (
+    EXTENSION_ALIAS_TO_TYPE_MAP,
+    EXTENSION_MONIKER_TO_ALIAS_MAP,
+    EXTENSION_TYPE_ACS,
+    EXTENSION_TYPE_OPS,
+    EXTENSION_TYPE_OSM,
+    EXTENSION_TYPE_PLATFORM,
+    EXTENSION_TYPE_SSC,
+    EXTENSION_TYPE_TO_MONIKER_MAP,
+    ClusterConnectStatus,
+)
+from azext_edge.edge.providers.orchestration.targets import InitTargets
+from azext_edge.edge.util import parse_kvp_nargs
+
+from ...generators import generate_random_string
+from .resources.conftest import (
+    BASE_URL,
+    CLUSTER_EXTENSIONS_API_VERSION,
+    CLUSTER_EXTENSIONS_URL_MATCH_RE,
+    CONNECTED_CLUSTER_API_VERSION,
+    get_base_endpoint,
+    get_mock_resource,
+)
+from .resources.test_instances_unit import (
+    get_instance_endpoint,
+    get_mock_cl_record,
+    get_mock_instance_record,
+)
+
+T = TypeVar("T", bound="UpgradeScenario")
+STANDARD_HEADERS = {"content-type": "application/json"}
+
+
+def get_mock_cluster_record(
+    resource_group_name: str,
+    name: str = "mycluster",
+    connected_status: str = ClusterConnectStatus.CONNECTED.value,
+) -> dict:
+    return get_mock_resource(
+        name=name,
+        properties={"connectivityStatus": connected_status},
+        resource_group_name=resource_group_name,
+    )
+
+
+def get_cluster_endpoint(resource_group_name: str, name: str = "mycluster") -> dict:
+    resource_path = "/connectedClusters"
+    if name:
+        resource_path += f"/{name}"
+    endpoint = get_base_endpoint(
+        resource_group_name=resource_group_name,
+        resource_path=resource_path,
+        resource_provider="Microsoft.Kubernetes",
+        api_version=CONNECTED_CLUSTER_API_VERSION,
+    )
+    endpoint = endpoint.replace("/resourceGroups/", "/resourcegroups/", 1)
+    return endpoint
+
+
+def get_cluster_extensions_endpoint(resource_group_name: str, cluster_name: str = "mycluster") -> dict:
+    resource_path = f"/connectedClusters/{cluster_name}/providers/Microsoft.KubernetesConfiguration/extensions"
+    return get_base_endpoint(
+        resource_group_name=resource_group_name,
+        resource_path=resource_path,
+        resource_provider="Microsoft.Kubernetes",
+        api_version=CLUSTER_EXTENSIONS_API_VERSION,
+    )
```

Comment:
resource group casing change only for base cluster?

Reply:
It is due to the rest api spec, which leads into the generated client, something I don't want to alter.